### PR TITLE
Improve saved exports UI

### PR DESCRIPTION
**Pull request by esoergel**
https://dimagi-dev.atlassian.net/browse/HI-300

That user was seeing an empty progress bar for days.  This turned out to be because of a simple bug (commit 1), however it was obscured because we used the `ignore_result` celery param, which ignores results, including failures (commit 3).  This also persisted beyond the 24 hour timeout (I believe) because the timeout is reset every time it's accessed, and with a browser tab open that happens every second or two (commit 2).

Another thing I noticed is that we direct users to refresh the page when the download is complete, but that's actually not necessary, they can just click the button (commit 5).

And finally I made use of the now-accurate `task.failed()` method to let the user know when the task fails and act appropriately.  The previous (unintentional) behavior was to freeze the progress wherever it was and stay there until no one looked at it for 24 hours.  Now it will display an error message and allow you to try again.

image

@orangejenny buddy @nickpell pinging 'cause I think you touched this recently too

### CODE REVIEW

File: corehq/apps/export/templates/export/partials/table.html [hunk 1]
```diff
@@ -188,10 +188,18 @@ <h4 data-bind="css: {'text-muted': hasEmailedExport && !isAutoRebuildEnabled()}"
...
+
+                            <p class="text-success" data-bind="if: emailedExport.taskStatus && emailedExport.taskStatus.justFinished() && emailedExport.taskStatus.success()">
                                 <i class="fa fa-check"></i>
                                 <strong>{% trans "Data update complete" %}</strong>
                             </p>
+
+                            <p class="text-danger" data-bind="if: emailedExport.taskStatus && emailedExport.taskStatus.justFinished() && emailedExport.taskStatus.failed()">
+                                <i class="fa fa-exclamation-triangle"></i>
+                                <strong>{% trans "Data update failed!" %}</strong>
+                                {% blocktrans %}If this problem persists, please <a href="#modalReportIssue" data-toggle="modal">Report an Issue</a>.{% endblocktrans %}
```
Comment (orangejenny) [inline corehq/apps/export/templates/export/partials/table.html]: Is there anything they should do? Should they click the Update Data button?

File: corehq/apps/export/static/export/js/export_list.js [hunk 2]
```diff
@@ -85,8 +90,9 @@ hqDefine("export/js/export_list", [
                         self.emailedExport.taskStatus.percentComplete(data.taskStatus.percentComplete);
                         self.emailedExport.taskStatus.started(data.taskStatus.started);
                         self.emailedExport.taskStatus.success(data.taskStatus.success);
+                        self.emailedExport.taskStatus.failed(data.taskStatus.failed);
                         self.emailedExport.taskStatus.justFinished(data.taskStatus.justFinished);
-                        if (!data.taskStatus.success) {
+                        if (!data.taskStatus.success & !data.taskStatus.failed) {
```
Comment (orangejenny) [inline corehq/apps/export/static/export/js/export_list.js]: `&&` not `&`

**Review by orangejenny — CHANGES_REQUESTED** [review]
just the typo

File: corehq/apps/export/templates/export/partials/table.html (hunk 1 above)
Comment (esoergel) [reply corehq/apps/export/templates/export/partials/table.html]: dunno that I'd recommend it - they'll probably spam that button anyways, but odds are it's a bug that'll fail every time

File: corehq/apps/export/static/export/js/export_list.js (hunk 2 above)
Comment (esoergel) [reply corehq/apps/export/static/export/js/export_list.js]: whoops, updated

**Review by orangejenny — APPROVED** [review]

**Pull request merged**